app/services/search_service.rb

```diff
@@ -0,0 +1,208 @@
+# The SearchService class is a utility class whose purpose is to provide detailed responses to queries within 
+# different categories (record types, functionality, subsystems, etc).
+# Though similar in operation to the TypeaheadService, the implementation is separate, in that the goal of the response
+# is to provide _detailed_ results at a deep level.  In effect, TypeaheadService provides pointers to 
+# better searches, while SearchService provides deep and detailed information.
+# TODO: Refactor TypeaheadService and SearchService so that common functions come from a higher level class?
```

Comment:
We recently saw Rails Concerns used to provide common methods to two related models -- this may only be viable for models: https://github.com/publiclab/plots2/tree/master/app/models/concerns but in general I like this idea.